src/openforms/contrib/brk/validators.py

```diff
@@ -0,0 +1,121 @@
+)
+@deconstructible
+class BRKZaakgerechtigdeValidator:
+
+    value_serializer = ValueSerializer
+
+    error_messages = {
+        "no_bsn": _("No BSN is available to validate your address."),
+        "retrieving_error": _(
+            "There was an error while retrieving the available properties. Please try again later"
+        ),
+        "not_owner": _("You are not one of the registered owners of this property"),
+        "multiple_cadastrals": _("We couldn't accurately determine your properties"),
+        "not_found": _("No property found for this address."),
+    }
+
+    def __call__(self, value: AddressValue, submission: Submission) -> bool:
+        try:
+            client = get_client()
+        except NoServiceConfigured as e:
+            raise ValidationError(self.error_messages["retrieving_error"]) from e
+
+        address_query: SearchParams = {
+            "postcode": value["postcode"],
+            "huisnummer": value["house_number"],
+        }
+        if "houseLetter" in value:
+            address_query["huisletter"] = value["house_letter"]
+        if "houseNumberAddition" in value:
+            address_query["huisnummertoevoeging"] = value["house_number_addition"]
+
+        # We assume submission has auth_info available, should we verify that this validator is used
+        # on a step that requires login?
+        if submission.auth_info.attribute != AuthAttribute.bsn:
+            raise ValidationError(self.error_messages["no_bsn"])
+
+        try:
+            with client:
+                cadastrals_resp = client.get_cadastrals_by_address(address_query)
+                kadastraals = glom(
+                    cadastrals_resp, "_embedded.kadastraalOnroerendeZaken", default=[]
+                )
+                if len(kadastraals) > 1:
+                    # The query by address returned more than one cadastral, this shouldn't happen
+                    raise ValidationError(self.error_messages["multiple_cadastrals"])
+                if not kadastraals:
+                    raise ValidationError(self.error_messages["not_found"])
+                kadastraal_id = kadastraals[0]["identificatie"]
+
+                titleholders_resp = client.get_cadastral_titleholders_by_cadastral_id(
+                    kadastraal_id
+                )
+
+                bsns = [
+                    a["persoon"]["identificatie"]
+                    for a in titleholders_resp["_embedded"]["zakelijkGerechtigden"]
+                ]
+                if submission.auth_info.attribute_hashed:
+                    is_valid = any(
+                        check_salted_hash(bsn, submission.auth_info.value)
+                        for bsn in bsns
+                    )
+                else:
+                    is_valid = submission.auth_info.value in bsns
+
+                if not is_valid:
+                    raise ValidationError(self.error_messages["not_owner"])
+
+        except (RequestException, KeyError, TypeError) as e:
```

Comment:
> I don't see any benefit of including `KeyError` and `TypeError` here.

Looking at it again it is a leftover from a previous solution I had without `glom`. The code is now safe, except here:
https://github.com/open-formulieren/open-forms/blob/ae52b85449d017785eac8c314734dd0b8d761758/src/openforms/contrib/brk/validators.py#L103-L106

but if we have exceptions with the above code, probably better to leave them unhandled as this would indicate something wrong from the BRK API.

Reply:
`try..catch` removed and ctx manager added

Reply:
I also tend to do the low-level processing on the API client classes, and let them return dataclasses suited to the data we're using/modelling. That also makes it type-checker friendlier. In a validator I don't really want to deal with the peculiarities of JSON+hal, that's the job of the API client who can translate it into domain objects.

I think the Haal Centraal/NP family members perform similar normalizations where it makes even more sense because there are different implementations that should also provide the same interface.